### ISSUE TITLE
[FEATURE] Upload Custom Image for Series / Book / Library

### ISSUE THREAD

**Issue by jacobkossman** [stumpapp/stump#202]
**Is your feature request related to a problem? Please describe.**
Currently you can only pick images from pages / book covers that already exist. It would be nice to allow image uploads (limited to jpg / png likely) for custom book, series, library images. There are cases where it'd be nice to have a custom cover that would not be available as a page from an existing book (eg abstract/minimalistic art, matching covers, etc)

**Describe the solution you'd like**
Allow the ability to upload a custom image for the various screens (book, series, library)

**Additional context**
Could also use this code for the user profile picture

**Comment by iwa** [comment]
would love to see this!

would it also be possible to allow webp images?

**Comment by aaronleopold** [comment]
> would love to see this!
> 
> 
> 
> would it also be possible to allow webp images?

Stump already supports the 3 common formats in the media processing areas of the code (JPG, PNG, WEBP). So, once I pick this ticket up, I'll make sure to support the same formats for uploads 🙂

**Comment by aaronleopold** [comment]
Wanted to give a quick update that I plan to rough out the upload aspect sometime this or next weekend (https://github.com/stumpapp/stump/issues/133). Once that is in place, I'll work on this 👍

**Comment by aaronleopold** [comment]
Not sure how much we care about GIF support, but it works locally with some tweaks

https://github.com/stumpapp/stump/assets/36278431/5e6ad37b-0d02-4b80-b0c6-73b11cb6561c

**Comment by aaronleopold** [comment]
Completed in https://github.com/stumpapp/stump/pull/249, thanks for the suggestion!
